src/perf_decode.ml

```diff
@@ -240,41 +256,96 @@ let ok_perf_line_to_event ?perf_maps matches line : Event.Ok.t =
         printf "Warning: skipping unrecognized perf output: %s\n%!" line;
         None
     in
-    { thread = { pid; tid }
-    ; time
-    ; data =
-        Trace
-          { trace_state_change
-          ; kind
-          ; src =
-              { instruction_pointer = src_instruction_pointer
-              ; symbol = src_symbol
-              ; symbol_offset = src_symbol_offset
-              }
-          ; dst =
-              { instruction_pointer = dst_instruction_pointer
-              ; symbol = dst_symbol
-              ; symbol_offset = dst_symbol_offset
-              }
-          }
-    }
+    Ok
+      { thread
+      ; time
+      ; data =
+          Trace
+            { trace_state_change
+            ; kind
+            ; src =
+                { instruction_pointer = src_instruction_pointer
+                ; symbol = src_symbol
+                ; symbol_offset = src_symbol_offset
+                }
+            ; dst =
+                { instruction_pointer = dst_instruction_pointer
+                ; symbol = dst_symbol
+                ; symbol_offset = dst_symbol_offset
+                }
+            }
+      }
   | results ->
     raise_s
       [%message "Regex of expected perf output did not match." (results : string array)]
 ;;
 
+let parse_perf_extra_sampled_event
+    ?perf_maps
+    (thread : Event.Thread.t)
+    time
+    period
+    line
+    lines
+    name
+    : Event.t
+  =
+  let (location : Event.Location.t) =
+    match lines with
+    | [] ->
+      (match Re.Group.all (Re.exec perf_extra_sampled_event_re line) with
+      | [| _str; _; instruction_pointer; symbol_and_offset |] ->
+        parse_location ?perf_maps ~pid:thread.pid instruction_pointer symbol_and_offset
+      | results ->
+        raise_s
+          [%message
+            "Regex of perf event did not match expected fields" (results : string array)])
+    | lines -> List.hd_exn lines |> parse_callstack_entry ?perf_maps thread
+  in
+  Ok
+    { thread
+    ; time
+    ; data = Event_sample { location; period = Option.value_exn period; name }
```

Comment:
If this is unconditionally going to raise on a `None` period, this should be reflected in the type signature (don't take an option).